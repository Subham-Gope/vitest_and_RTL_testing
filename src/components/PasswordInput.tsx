import React, { useState } from "react";
import type { ChangeEvent } from "react";

const PasswordInput: React.FC = () => {
  const [pass, setPass] = useState<string>("");

  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setPass(e.target.value);
  };

  return (
    <>
      <label htmlFor="pass">Enter Your Password</label>
      <input
        type="text"
        placeholder="jbji#Ib3j2"
        id="pass"
        name="pass"
        value={pass}
        onChange={handleChange}
      />
      {pass && <p>Password: {pass}</p>}
    </>
  );
};

export default PasswordInput;
