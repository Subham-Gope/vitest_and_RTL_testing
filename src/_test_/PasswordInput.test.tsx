import { describe, test, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import PasswordInput from "../components/PasswordInput";

describe("Testing the PasswordInput.tsx", () => {
  test("registered a pass word input and displays it", async () => {
    // Setting up the userEvent
    const user = userEvent.setup();

    const { container } = render(<PasswordInput />);
    expect(container).toMatchSnapshot();

    const input = screen.getByLabelText(/^enter your password$/i);
    expect(input).toBeInTheDocument();

    // Initiating the user interactions
    await user.type(input, "ghfjr3@JDK");
    expect(input).toHaveValue("ghfjr3@JDK");

    const displayText = screen.getByText(/^password:$/i);
    expect(displayText).toBeInTheDocument();
    
  });
});
