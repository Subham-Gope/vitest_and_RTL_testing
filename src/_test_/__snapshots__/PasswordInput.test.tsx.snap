// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the PasswordInput.tsx > registered a pass word input and displays it 1`] = `
<div>
  <label
    for="pass"
  >
    Enter Your Password
  </label>
  <input
    id="pass"
    name="pass"
    placeholder="jbji#Ib3j2"
    type="text"
    value=""
  />
</div>
`;
